include/pcp/filter/filter.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef PCP_FILTER_FILTER_HPP
```

Comment:
Since you have added a new "filter" module, you need to add the corresponding #include directive in the top-level pcp.hpp file.